tests/serialization/test_filter.py

```diff
@@ -66,3 +69,46 @@ def test_filter_array_like_categoricals(op, expected, cat_type):
     res = filter_array_like(ser, op, "B")
 
     assert all(res == expected)
+
+
+@pytest.mark.parametrize(
+    "value, filter_value",
+    [
+        (1, 1.0),
+        ("1", 1.0),
+        ("1", 1),
+        (1, "1"),
+        (datetime.date(2019, 1, 1), 1),
+        (datetime.datetime(2019, 1, 1), 1),
+        (datetime.datetime(2019, 1, 1), datetime.date(2019, 1, 1)),
+        (datetime.date(2019, 1, 1), datetime.datetime(2019, 1, 1)),
+    ],
+)
+@pytest.mark.parametrize("op", ["==", "!=", "<", "<=", ">", ">="])
+def test_raise_on_type(value, filter_value, op):
+    array_like = pd.Series([value])
+    with pytest.raises(TypeError, match="Unexpected type encountered."):
+        filter_array_like(array_like, op, filter_value, strict_date_types=True)
+
+
+@pytest.mark.parametrize("op", ["==", "!=", ">=", "<=", ">", "<"])
+def test_filter_df_from_predicates(op):
+    df = pd.DataFrame(
+        {
+            "A": range(10),
+            "B": ["A", "B"] * 5,
+            "C": pd.Series(["X", "Y"] * 5).astype("category"),
+            "D": pd.Series([datetime.date(2019, 1, 1), datetime.date(2019, 1, 2)] * 5),
+            "E": [datetime.datetime(2019, 1, 1), datetime.datetime(2019, 1, 2)] * 5,
+        }
+    )
+
+    ix = 4
+    for col in df.columns:
```

Comment:
same, I would use a `parametrize` here as well, because it simplifies the debugging in case of a test failure.